nifi-standard-bundle/nifi-standard-processors/src/main/java/com/asymmetrik/nifi/processors/FetchFileSplits.java

```diff
@@ -0,0 +1,389 @@
+@CapabilityDescription("Reads in a file from the local filesystem and splits it while reading it in.")
+public class FetchFileSplits extends AbstractProcessor {
+
+    static final AllowableValue COMPLETION_NONE = new AllowableValue("None", "None", "Leave the file as-is");
+    static final AllowableValue COMPLETION_MOVE = new AllowableValue("Move File", "Move File", "Moves the file to the directory specified by the <Move Destination Directory> property");
+    static final AllowableValue COMPLETION_DELETE = new AllowableValue("Delete File", "Delete File", "Deletes the original file from the file system");
+
+    static final AllowableValue CONFLICT_REPLACE = new AllowableValue("Replace File", "Replace File", "The newly ingested file should replace the existing file in the Destination Directory");
+    static final AllowableValue CONFLICT_KEEP_INTACT = new AllowableValue("Keep Existing", "Keep Existing", "The existing file should in the Destination Directory should stay intact and the newly "
+            + "ingested file should be deleted");
+    static final AllowableValue CONFLICT_FAIL = new AllowableValue("Fail", "Fail", "The existing destination file should remain intact and the incoming FlowFile should be routed to failure");
+    static final AllowableValue CONFLICT_RENAME = new AllowableValue("Rename", "Rename", "The existing destination file should remain intact. The newly ingested file should be moved to the "
+            + "destination directory but be renamed to a random filename");
+
+
+    static final Relationship REL_ORIGINAL = new Relationship.Builder()
+        .name("original")
+        .description("Upon successful completion of the file fetch and split, the " +
+                "original FlowFile is routed to 'original'")
+        .build();
+    static final Relationship REL_SPLITS = new Relationship.Builder()
+        .name("splits")
+        .description("The splits of the input file are routed to 'splits'")
+        .build();
+    static final Relationship REL_NOT_FOUND = new Relationship.Builder()
+        .name("not.found")
+        .description("Any FlowFile that could not be fetched from the file system because the file could not be found will be transferred to this Relationship.")
+        .build();
+    static final Relationship REL_PERMISSION_DENIED = new Relationship.Builder()
+        .name("permission.denied")
+        .description("Any FlowFile that could not be fetched from the file system due to the user running NiFi not having sufficient permissions will be transferred to this Relationship.")
+        .build();
+    static final Relationship REL_FAILURE = new Relationship.Builder()
+        .name("failure")
+        .description("If any errors occur during reading or splitting, the " +
+                "original flow file is routed to 'failure'")
+        .build();
+
+    static final PropertyDescriptor LINES_TO_SKIP = new PropertyDescriptor.Builder()
+        .name("Lines to Skip")
```

Comment:
name property should be more like an ID.  The displayName property is more human readable.  The reason for this is that the name property cannot be changed without breaking any processors already in use.  The displayName property, on the other hand, can be modified without causing any issues.

Reply:
Good call - I'll change that now.